pkg/internal/sdk/template/brc29/brc29_address_test.go

```diff
@@ -0,0 +1,152 @@
+package brc29_test
+
+import (
+	"testing"
+
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/internal/sdk"
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/internal/sdk/template/brc29"
+	ec "github.com/bsv-blockchain/go-sdk/primitives/ec"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAddress(t *testing.T) {
+	errorTestCases := map[string]struct {
+		sender    string
+		keyID     brc29.KeyID
+		recipient string
+	}{
+		"return error when sender key is empty": {
+			sender:    "",
+			keyID:     keyID,
+			recipient: invalidKeyHex,
+		},
+		"return error when sender key parsing fails": {
+			sender:    invalidKeyHex,
+			keyID:     keyID,
+			recipient: recipientPublicKeyHex,
+		},
+		"return error when KeyID is invalid": {
+			sender:    senderPrivateKeyHex,
+			keyID:     brc29.KeyID{DerivationPrefix: "", DerivationSuffix: ""},
+			recipient: recipientPublicKeyHex,
+		},
+		"return error when recipient key is empty": {
+			sender:    senderPrivateKeyHex,
+			keyID:     keyID,
+			recipient: "",
+		},
+		"return error when recipient key parsing fails": {
+			sender:    senderPrivateKeyHex,
+			keyID:     keyID,
+			recipient: invalidKeyHex,
+		},
+	}
+	for name, test := range errorTestCases {
+		t.Run(name, func(t *testing.T) {
+			address, err := brc29.Address(test.sender, test.keyID, test.recipient)
+
+			require.Nil(t, address)
+			require.Error(t, err)
+		})
+	}
+
+	t.Run("return error when nil is passed as sender private key deriver", func(t *testing.T) {
+		var keyDeriver *sdk.KeyDeriver
+
+		address, err := brc29.Address(keyDeriver, keyID, recipientPublicKeyHex)
+
+		assert.Error(t, err)
+		require.Nil(t, address)
+	})
+
+	t.Run("return error when nil is passed as sender private key", func(t *testing.T) {
+		var priv *ec.PrivateKey
+
+		address, err := brc29.Address(priv, keyID, recipientPublicKeyHex)
+
+		assert.Error(t, err)
```

Comment:
```suggestion
		require.Error(t, err)
```
this is minor thing here, but slightly inconsistent with the test fail-fast .. would u agree ?

Reply:
same thing as here:
https://github.com/bsv-blockchain/go-wallet-toolbox/pull/275#discussion_r2139542316